### ISSUE TITLE
[Enhancement] Popups and actions should be more decoupled

### ISSUE THREAD

**Issue by asmodeus812** [NeogitOrg/neogit#1446]
It is sometimes nice to be able to use and trigger a specific action that a popup provides directly, for example open the branches poupup and trigger the `c` checkout action, programatically. As of now the solution is to call popup create and then feedkeys. This is somewhat cumbersome, it would be great the actions do not rely on a popup argument to be passed in by default, some actions (in branch actions for example) rely in some state from popup which makes it hard to reuse these actions outside of popups. Another option is to allow us to `create` popups without showing them or in other words - not calling the `show` method, (note, care should be appllied to avoid building any context related to showing on screen, such as creating a new buffer, lines etc, since it will never be shown, the poup object would simply act as a wrapper around the actions to be used only programatically, so in case any resources related to `showing` the popup are created outside the `popup.show` method, those should be pruned if possible), and expose an api that would allow us to trigger the actions from the popup api. Further more besides actions, one should be able to also programatically set the flags/switches.

**Comment by CKolkey** [comment]
I think this already exists, no? https://github.com/NeogitOrg/neogit/blob/master/doc/neogit.txt#L582-L607

Admittedly the popup state/context table passed into the constructor isn't optional, like you stipulate (and I think you're right that it should be).

**Comment by asmodeus812** [comment]
Yeah i can check them out, had a quick look at all popups, most are verifying against existing properties on state.env, only the bisect popup is a bit complex does not seem to have any fallback logic at all.

**Comment by sep2** [comment]
I got a hacky way to work around this if anyone is interested https://github.com/sep2/nvim/blob/757a6e90421aafe517ba37ed879fed8c9954b1f2/lua/plugins/git.lua#L36-L50

**Comment by CKolkey** [comment]
Nice! But there's a less hacky way too: https://github.com/NeogitOrg/neogit/blob/master/lua/neogit.lua#L182-L232

I suppose this issue can be closed :)
